ktlint-line-formatter/src/main/kotlin/com/pinterest/ktlint/LineFormatter/LineParse.kt

```diff
@@ -0,0 +1,46 @@
+package com.pinterest.ktlint.LineFormatter
+
+class LineSyntaxCheck {
+
+    fun HasCommentAtEnd(ln: String) : Int {
```

Comment:
I did a quick look at previous commits to the `ktlint` project to see how we should traverse the syntax tree. Based on https://github.com/pinterest/ktlint/commit/b550d1807e4a614bae4f60b25dc5fb9027d8dd9b, I think it'd be better to use the existing existing abstract syntax tree classes:
```

import com.github.shyiko.ktlint.core.ast.ElementType.RPAR
import com.github.shyiko.ktlint.core.ast.ElementType.SAFE_ACCESS
import com.github.shyiko.ktlint.core.ast.ElementType.SAFE_ACCESS_EXPRESSION
import com.github.shyiko.ktlint.core.ast.ElementType.SUPER_TYPE_CALL_ENTRY
import com.github.shyiko.ktlint.core.ast.ElementType.SUPER_TYPE_ENTRY
import com.github.shyiko.ktlint.core.ast.ElementType.SUPER_TYPE_LIST
import com.github.shyiko.ktlint.core.ast.ElementType.THEN
...
```

Those will make these checks more extensible as new formatting cases come up.